### PR TITLE
github-merge: warn if ack message contains @

### PR DESCRIPTION
**Pull request by jonatack**
for cases like https://github.com/bitcoin/bitcoin/pull/17925#issuecomment-579946380.

### CODE REVIEW

File: github-merge.py [hunk 1]
```diff
@@ -195,7 +195,10 @@ def print_merge_details(pull, title, branch, base_branch, head_branch, acks, mes
         if acks:
             print('{}ACKs:{}'.format(ATTR_PR, ATTR_RESET))
             for name, message in acks.items():
-                print('* {} {}({}){}'.format(message, ATTR_NAME, name, ATTR_RESET))
+                if message is not None and '@' in message:
```
Comment (fanquake) [inline github-merge.py]: Shouldn't this just print an extra line in the @ case? Right now it'll print out the warning, but you wont see the offending comment (which could be fine when containing @).

**Comment by emilengler** [comment]
Why not use a regex to check if a username follows the @?
Maybe some people use a non tagging @ like
```
Hello @ jonatack
```
Maybe try `r"@[\s]{1,}"`

File: github-merge.py (hunk 1 above)
Comment (jonatack) [reply github-merge.py]: Yes, I thought of that but began with simply doing the same thing as 6 lines down when the message contains an @.

What I'm not quite clear on is, the acks seem to already be covered by the existing '@'-check?
```python
acks = get_acks_from_comments(head_commit=head_commit, comments=comments)
message += make_acks_message(head_commit=head_commit, acks=acks)
.../...
print_merge_details(pull, title, branch, base_branch, head_branch, acks, message)
```

Comment (laanwj) [reply github-merge.py]: Right—I'd prefer to print the comment as normal, store a flag and print a single warning at the end.

**Comment by laanwj** [comment]
concept ACK, thanks for working on this

**Comment by fanquake** [comment]
> Why not use a regex to check if a username follows the @?

NACK. That's completely unnecessary for what this script needs to achieve.

File: github-merge.py (hunk 1 above)
Comment (jonatack) [reply github-merge.py]: Another possibility would be to tighten up the ACKs checking in `get_acks_from_comments`, but as a first step ensuring they are warned about seemed more prudent.

Comment (jonatack) [reply github-merge.py]: > Right—I'd prefer to print the comment as normal, store a flag and print a single warning at the end.

Right -- will update

**Comment by jonatack** [comment]
One sec: the use of message as both an argument and local var seems to have disabled the message check.

**Comment by jonatack** [comment]
The use of message as both an argument and local var seems to have disabled the message check. Depending on how this script is being used, the added check might not be needed.

**Comment by jonatack** [comment]
Can squash after feedback on what is preferred.

**Comment by laanwj** [comment]
Needs rebase after #53.

File: github-merge.py [hunk 2]
```diff
@@ -191,15 +191,20 @@ def make_acks_message(head_commit, acks) -> str:
...
+            for name, msg in acks.items():
+                if msg is not None and '@' in msg:
+                    warnings += 1
+                print('* {} {}({}){}'.format(msg, ATTR_NAME, name, ATTR_RESET))
         else:
             print('{}Top commit has no ACKs!{}'.format(ATTR_WARN, ATTR_RESET))
     if message is not None and '@' in message:
         print('{}Merge message contains an @!{}'.format(ATTR_WARN, ATTR_RESET))
+    if warnings > 0:
+        print('{}{} merge ACK message(s) contain(s) an @!{}'.format(ATTR_WARN, warnings, ATTR_RESET))
```
Comment (MarcoFalke) [inline github-merge.py]: The message should already contain the ACKs, thus should already warn about that, no?

https://github.com/bitcoin-core/bitcoin-maintainer-tools/blob/35eb2c71d09993eb7e87febec0e08aec347b8e7c/github-merge.py#L375-L386

**Comment by jonatack** [comment]
> The use of `message` as both an argument and local var in `print_merge_details()` seems to mess with the @-in-message check at the end. Depending on how this script is being used, once fixed, the additional check might not be needed.

Rebased. The bugfix in this PR was added to #53. Unsure if this PR is still useful.

File: github-merge.py (hunk 2 above)
Comment (jonatack) [reply github-merge.py]: The main fix in this PR (the first commit) was added to #53. I don't know if the additional checks are useful now.

**Pull request closed without merging**